src/backend/distributed/planner/modify_planner.c

```diff
@@ -544,6 +568,86 @@ DistributedModifyShardInterval(Query *query)
 
 
 /*
+ * UseFastShardPruning returns true if the commandType is INSERT and partition method
+ * is hash or range.
+ */
+static bool
+UseFastShardPruning(CmdType commandType, char partitionMethod)
+{
+	/* we currently only support INSERTs */
+	if (commandType != CMD_INSERT)
+	{
+		return false;
+	}
+
+	/* INSERTs could only go into hash and range distributed tables */
```

Comment:
Easy to misunderstand comment - if that's the case, there's no need for an if... Maybe say that's currently the case?